pip/src/interpreter.rs

```diff
@@ -336,24 +337,25 @@ impl Output {
         self.__repr__()
     }
 
-    fn _repr_html_(&self) -> String {
+    fn _repr_markdown_(&self) -> Option<String> {
```

Comment:
Why Markdown in favor of HTML? Does it have implications in how the cell output format is selected VS Code? I seem to recall there was some pretty funky UX going on there. Am I correct in assuming that you're trying to simplify the UX here by having just one "rich" output format instead of latex/html etc?

Reply:
Also, do you know if JupyterLab renders these outputs correctly?

Reply:
You can embed LaTeX directly in Markdown and it renders correctly - not so in HTML. Being that the other HTML outputs also render OK as Markdown (as inline HTML), this enabled the 'pretty printing' of the equation as well as the DumpMachine table.

Reply:
Yeah, what Bill said. HTML won't render embedded LaTeX, but Markdown renders both HTML and LaTeX, making it the one ring to rule them all. Seemed like a good reason to drop the other representations and align behind Markdown as a single rich output format.

I did get the chance to test all our output variants (messages, state dumps from `DumpMachine` and`DumpRegister`, and matrices from `DumpOperation`) in VS Code notebooks, Jupyter in the browser, and JupyterLab in the browser, and confirmed they are all rendered as expected across the board. Bill identified some quirks of the browser formatting that we worked around to make sure it looked good and didn't do anything odd.